app/main.py

```diff
@@ -1,3 +1,89 @@
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        if self._is_number(other):
+            return Distance(km=self.km + other)
+        else:
+            raise TypeError("Unsupported operand type for + operator")
+
+    def __iadd__(self, other: Any) -> "Distance":
+        if self._is_number(other):
+            self.km += other
+            return self
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        else:
+            raise TypeError("Unsupported operand type for += operator")
+
+    def __mul__(self, other: Any) -> "Distance":
+        if self._is_number(other):
+            return Distance(km=self.km * other)
+        else:
+            raise TypeError("Unsupported operand type for * operator")
+
+    def __truediv__(self, other: Any) -> "Distance":
+        if isinstance(other, int | float):
```

Comment:
The use of `int | float` for type checks is not compatible with Python versions prior to 3.10. To ensure broader compatibility, please use `isinstance(other, (int, float))`.